sklearn/feature_selection/_univariate_selection.py

```diff
@@ -986,6 +991,93 @@ def _get_support_mask(self):
         return self.pvalues_ < self.alpha / len(self.pvalues_)
 
 
+class SelectThreshold(_BaseFilter):
+    """Select features by keeping the ones having a score higher than a threshold.
+
+    Read more in the :ref:`User Guide <univariate_feature_selection>`.
+
+    .. versionadded:: 1.4
+
+    Parameters
+    ----------
+    score_func : callable, default=f_classif
+        Function taking two arrays `X` and `y`, and returning a pair of arrays
+        (scores, pvalues) or a single array with scores.
+        Default is :func:`f_classif` (see below "See Also"). The default function only
+        works with classification tasks. In an unsupervised setting, you can set
+        `y=None` in the signature of `score_func`.
+
+    threshold : float, default=0.5
+        Score threshold above which a feature is selected.
+
+    Attributes
+    ----------
+    scores_ : array-like of shape (n_features,)
+        Scores of features.
+
+    pvalues_ : array-like of shape (n_features,)
+        p-values of feature scores, None if `score_func` returned only scores.
```

Comment:
it makes me cringe that we're reporting p-values here 😁 and that there's no comment on whether they're corrected or not, and how they should be corrected if not, before being potentially used.